src/Diagnostics.DataProviders/DataProviders.cs

```diff
@@ -15,6 +16,7 @@ public class DataProviders
         public IChangeAnalysisDataProvider ChangeAnalysis;
         public IAscDataProvider Asc;
         public Func<MdmDataSource, IMdmDataProvider> Mdm;
+        public Func<IMdmDataProviderConfiguration, IMdmDataProvider> Mdm2;
```

Comment:
Maybe we can come up with a better name for this to avoid confusion

Reply:
@nmallick1 what do you suggest?

Reply:
Renamed to MdmGeneric